ignite/distributed/comp_models/native.py

```diff
@@ -450,7 +450,7 @@ def _expand_hostlist(nodelist: str) -> List[str]:
 
         result_hostlist = []
         for node in node_list:
-            nodelist_match = r"(\w+-?)\[((,?[0-9]+-?,?-?){0,})\](.*)?"
+            nodelist_match = r"(.+)\[((,?[0-9]+-?,?-?){0,})\](.*)?"
             if re.search(nodelist_match, node):
```

Comment:
I also found below code a bit strange:
```python
        if re.search(nodelist_match, node):
            match = re.search(nodelist_match, node)

            if match is None:
                raise ValueError(f"hostlist unvalid : {nodelist}")
```
why can't we do it like 
```python
        match = re.search(nodelist_match, node)
        if match is not None:            
```

Reply:
Sorry for this lack of quality !